src/state/transfersSelectors.ts

```diff
@@ -0,0 +1,39 @@
+import { DydxAddress } from '@/constants/wallets';
+
+import { RootState } from './_store';
+import { createAppSelector } from './appTypes';
+import { Deposit } from './transfers';
+
+const getTransfersByAddress = (state: RootState) => state.transfers.transfersByDydxAddress;
+
+export const selectPendingDeposits = () =>
+  createAppSelector(
+    [getTransfersByAddress, (s, dydxAddress?: DydxAddress) => dydxAddress],
+    (transfersByAddress, dydxAddress): Deposit[] => {
+      if (!dydxAddress || !transfersByAddress[dydxAddress]) return [];
+
+      return transfersByAddress[dydxAddress].filter(
+        (transfer) => transfer.type === 'deposit' && transfer.status === 'pending'
+      ) as Deposit[];
```

Comment:
is there a way to do this without the cast? Not sure if we could utilize a `transfer is Deposit` filter function :/

Reply:
that works for individual `transfer` items but not for the filtered array itself :/ i like the idea of adding some explicit type guard functions tho